nbgrader/docs/source/configuration/jupyterhub_config.rst

```diff
@@ -195,3 +200,135 @@ if students may be enrolled in multiple courses, you need to provide them a
 custom ``nbgrader_config.py`` that will cause nbgrader to look for assignments
 in a subdirectory corresponding to the course name. See :ref:`multiple-classes`
 for details.
+
+
+
+JupyterHub Authentication
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+
+With the advent of JupyterHubAuthPlugin students who don't have a specified course_id won't see all courses anymore, just the ones they have been added to. 
+
+Requirements for using ``JupyterHubAuthPlugin``: 
+
+* Jupyterhub > 0.8
+* Activating the JupyterHubAuthPlugin requires you to add it as an authentication plugin class into ``/etc/jupyter/nbgrader_config.py``
+.. code:: python
+
+    from nbgrader.auth import JupyterHubAuthPlugin
+    c = get_config()
+    c.Authenticator.plugin_class = JupyterHubAuthPlugin
+
+* Instructor and student groups need to be named ``formgrade-{course_id}`` for instructors and ``nbgrader-{course_id}`` for students
+* The course service needs to have an api_token set that is from a jupyterhub admin see: Creating an `api token <https://jupyterhub.readthedocs.io/en/stable/reference/rest.html#create-an-api-token>`_
+
+
+* The service user also needs to be added to the formgrade group (see the jupyterhub_config example below)
+
+
+As in the case of multiple graders for a single class, if you have multiple
+classes on the same JupyterHub instance, then you will need to create multiple
+services (one for each course) and corresponding accounts for each service
+(with the nbgrader extensions enabled, see :doc:`/user_guide/installation`).
+For example, you could have users ``grader-course101`` and
+``grader-course123``. Your JupyterHub config would then look something like
+this:
+
+.. code:: python
+
+    c = get_config()
+
+    # Our user list
+    c.Authenticator.whitelist = [
```

Comment:
I think `grader-course101` and `grader-course123` need to be added to the whitelist as well.

Reply:
Added to the whitelist